core/logic/src/main/java/org/apache/syncope/core/logic/TaskLogic.java

```diff
@@ -459,8 +462,9 @@ public void actionJob(final String key, final JobAction action) {
             final Date since,
             final List<ExecStatus> statuses,
             final List<String> resources) {
-        return taskDAO.purgePropagations(since, statuses,
-                resources.stream().map(resourceDAO::find).filter(Objects::nonNull).collect(Collectors.toList()));
+        return taskDAO.purgePropagations(since, statuses, resources == null 
+                ? Collections.<ExternalResource>emptyList()
```

Comment:
why `Collections.<ExternalResource>emptyList()` and not simply `null`? There is already a check in the DAO

Reply:
yes, it's better